src/mokernels/lmm.jl

```diff
@@ -0,0 +1,46 @@
+@doc raw"""
+    NaiveLMMMOKernel(g, e::MOKernel, A::AbstractMatrix)
+
+Kernel associated with the linear mixing model.
+
+# Definition
+
+For inputs ``x, x'`` and output dimensions ``p_x, p_{x'}'``, the kernel is defined as[^BPTHST]
+```math
+k\big((x, p_x), (x, p_{x'})\big) = H_{p_{x}}K(x, x')H_{p_{x'}}
+```
+where ``K(x, x') = Diag(k_1(x, x'), ..., k_m(x, x'))`` with zero off-diagonal entries.
+``H_{p_{x}}`` is the ``p_x``-th column of ``H`` and
+``k_1, \ldots, k_m`` are ``m`` kernels, one for each latent process, ``H`` is a
+mixing matrix of ``m`` basis vectors spanning the output space.
+
+[^BPTHST]: Wessel P. Bruinsma, Eric Perim, Will Tebbutt, J. Scott Hosking, Arno Solin, Richard E. Turner (2020). [Scalable Exact Inference in Multi-Output Gaussian Processes](https://arxiv.org/pdf/1911.06287.pdf).
+"""
+struct NaiveLMMMOKernel{Tk<:AbstractVector{<:Kernel}, Th<:AbstractMatrix} <: MOKernel
+    K::Tk
+    H::Th
+    # if just a simple kernel is provided, we construct the MO kernel
+    function NaiveLMMMOKernel(k::SimpleKernel, H::AbstractMatrix)
+        m = size(H,2)
+        K = fill(k, m)
+        return new{typeof(K),typeof(H)}(K,H)
+    end
+    function NaiveLMMMOKernel(K::AbstractVector{<:Kernel}, H::AbstractMatrix)
+        return new{typeof(K),typeof(H)}(K,H)
+    end
+end
+
+function (κ::NaiveLMMMOKernel)((x, px)::Tuple{Any,Int}, (y, py)::Tuple{Any,Int})
+    (px > size(κ.H, 2) || py > size(κ.H, 2) || px < 1 || py < 1) &&
+    error("`px` and `py` must be within the range of the number of outputs")
+    return sum(H[i, px] * κ.K[i](x, y) * H[i, py] for i in 1:length(κ.K))
+end
+
+function Base.show(io::IO, k::NaiveLMMMOKernel)
+    return print(io, "Linear Mixing Model Multi-Output Kernel (naive implementation)")
+end
+
+function Base.show(io::IO, ::MIME"text/plain", k::NaiveLMMMOKernel)
+    print(io, "Linear Mixing Model Multi-Output Kernel (naive implementation)\n\tkernels (K): ")
+    return join(io, k.K, "\n\t\t")
+end
```

Comment:
I don't think this is quite right as-is. e.g.
```jl
julia> lmm = KernelFunctions.NaiveLMMMOKernel([Matern32Kernel(), SqExponentialKernel()], rand(2,3))
Linear Mixing Model Multi-Output Kernel (naive implementation)
        kernels (K): Matern 3/2 Kernel (metric = Distances.Euclidean(0.0))
                Squared Exponential Kernel (metric = Distances.Euclidean(0.0))
```
perhaps we want something like
```jl
Linear Mixing Model Multi-Output Kernel (naive implementation). Kernels:
        Matern 3/2 Kernel (metric = Distances.Euclidean(0.0))
        Squared Exponential Kernel (metric = Distances.Euclidean(0.0))
```
Also, we should probably call `show` on the kernels, rather than implicitly string-ifying them via `join`:
```suggestion
function Base.show(io::IO, mime::MIME"text/plain", k::NaiveLMMMOKernel)
    print(io, "Linear Mixing Model Multi-Output Kernel (naive implementation). Kernels:")
    for k in k.K
        print(io, "\n\t")
        show(io, mime, k)
    end
end
```

Reply:
> than implicitly string-ifying them via join:

Does `join(io, ...)` actually create a string first before writing it to `io`? I assumed it would just write to `io` directly.

Reply:
Done :)

Reply:
Is everyone happy with its current state?

Reply:
> Does join(io, ...) actually create a string first before writing it to io?

i looked into it and no, it just calls `print(io, ...)` which writes straight to `io`. 

But what i meant to say is that we probably want to call 3-arg `show` on the inner kernels, rather than using `join`/`print` which are 2-arg show.

Sometimes words are harder that code 😂 sorry for the confusion. 

Let us know if you think it should be as it was before, @devmotion 

(p.s. `show`/`print` stuff is some of the most confusing stuff in all Julia-land 😵)

Reply:
I will assume this is good as is (for the sake of this PR)